app/models/piece.rb

```diff
@@ -1,3 +1,91 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def current_pieces_coords
+    current_pieces_coords = []
+    self.game.pieces.map { |piece| current_pieces_coords << [piece.x_coord, piece.y_coord] }
```

Comment:
@Brandongoodman615 the whole method can be just one line :) since we store exactly the same array in a variable and return the same variable..
```
self.game.pieces.map { |piece| current_pieces_coords << [piece.x_coord, piece.y_coord] }
```